libs/tdigest/src/main/java/org/elasticsearch/tdigest/SortingDigest.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.tdigest;
+
+import java.util.AbstractCollection;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * Simple implementation of the TDigest interface that stores internally and sorts all samples to calculate quantiles and CDFs.
+ * It provides perfect quantile and cdf calculations and matches or exceeds the performance of MergingDigest at least for millions of
+ * samples, at the expense of allocating much more memory.
+ */
+public class SortingDigest extends TDigest {
```

Comment:
I think this is a good idea 👍 . However can we add this class, `MergingTDigest` and related code via a separate change? This PR is already large and just keeping this PR to do bare minimum I think would be easier for the review process.